packages/ra-core/src/controller/field/useReference.ts

```diff
@@ -80,6 +83,8 @@ export const useReference = ({
     const rootPath = basePath.replace(resource, reference);
     const resourceLinkPath = !linkType
         ? false
+        : typeof linkType === 'function'
+        ? linkType()
```

Comment:
what's the use of passing a function if that function doesn't have access to the `record` and the `referenceRecord`?